lib/vtls/schannel.c

```diff
@@ -383,7 +402,7 @@ schannel_connect_step1(struct connectdata *conn, int sockindex)
   }
   memset(connssl->ctxt, 0, sizeof(struct curl_schannel_ctxt));
 
-  host_name = Curl_convert_UTF8_to_tchar(hostname);
+  host_name = Curl_convert_UTF8_to_tchar(conn->host.name);
```

Comment:
Good catch, this change has been reverted.

My changes were originally developed before HTTPS proxy support landed in master. When I was rebasing these changes onto master, I ran into some conflicts. In the process of resolving those changes, I think I inadvertently made this change.